bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java

```diff
@@ -0,0 +1,363 @@
+                httpClient);
+
+        // check access and pair if necessary
+        if (!httpClient.isAccessPossible()) {
+            // update status already if access is not possible
+            this.updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.UNKNOWN.NONE, "@text/offline.conf-error-pairing");
+            if (!httpClient.doPairing()) {
+                this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "@text/offline.conf-error-pairing");
+            }
+            // restart initial access - needed also in case of successful pairing to check access again
+            scheduleInitialAccess(httpClient);
+        } else {
+            // print rooms and devices if things are reachable
+            boolean thingReachable = true;
+            thingReachable &= this.getRooms();
+            thingReachable &= this.getDevices();
+
+            if (thingReachable) {
+                this.updateStatus(ThingStatus.ONLINE);
+
+                // Start long polling
+                try {
+                    this.longPolling.start(httpClient);
+                } catch (LongPollingFailedException e) {
+                    this.handleLongPollFailure(e);
+                }
+            } else {
+                this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        "@text/offline.not-reachable");
+                // restart initial access
+                scheduleInitialAccess(httpClient);
+            }
+        }
+    }
+
+    /**
+     * Get a list of connected devices from the Smart-Home Controller
+     */
+    private boolean getDevices() {
+        BoschHttpClient httpClient = this.httpClient;
+        if (httpClient == null) {
+            return false;
+        }
+
+        try {
+            logger.debug("Sending http request to Bosch to request clients: {}", config.ipAddress);
+            String url = httpClient.createSmartHomeUrl("devices");
+            ContentResponse contentResponse = httpClient.createRequest(url, GET).send();
+
+            String content = contentResponse.getContentAsString();
+            logger.debug("Response complete: {} - return code: {}", content, contentResponse.getStatus());
+
+            Type collectionType = new TypeToken<ArrayList<Device>>() {
+            }.getType();
+            ArrayList<Device> devices = gson.fromJson(content, collectionType);
+
+            if (devices != null) {
+                for (Device d : devices) {
+                    // Write found devices into openhab.log until we have implemented auto discovery
+                    logger.info("Found device: name={} id={}", d.name, d.id);
+                    if (d.deviceSerivceIDs != null) {
+                        for (String s : d.deviceSerivceIDs) {
+                            logger.info(".... service: {}", s);
```

Comment:
either remove this log or lower the logging level.

Reply:
This is meant as a workaround until we have the auto discovery implemented. It will log the available devices with their IDs. The users need those IDs to create the things, so we like to keep it at this log level for the moment. Otherwise the users would have to increase the log level to find this information which might be a problem for not so advanced users.

Reply:
@cpmeister Is my explanation understandable? Can we keep it in?

Reply:
You can keep it in for now.